source/common/tls/cert_validator/default_validator.h

```diff
@@ -111,10 +112,12 @@ class DefaultCertValidator : public CertValidator, Logger::Loggable<Logger::Id::
                                  Envoy::Ssl::ClientValidationStatus& detailed_status,
                                  std::string* error_details, uint8_t* out_alert);
 
+  const std::string cert_name_;
   const Envoy::Ssl::CertificateValidationContextConfig* config_;
   SslStats& stats_;
   Server::Configuration::CommonFactoryContext& context_;
-
+  Stats::Scope& scope_;
```

Comment:
I think you can create the stat in the constructor, and then not need to keep a scope reference

Reply:
Good point. Removed

Reply:
side note: for spiffe validator we'll keep the initializeCertExpirationStats call in the constructor since that's where the logic to populate the cert is located.

Reply:
Pass the scope reference as a parameter to initializeSslContexts` and then you don't need to hold the reference ever.

Reply:
Done, removed storing references to scope_